tools/check.py

```diff
@@ -87,343 +80,421 @@ def _validate_no_fixme(self):
-        * Checks # times callout style appears in document, minc <= n <= maxc
-        """
-        heading_node = callout_node.children[0]
-        valid_head_lvl = self.ast.is_heading(heading_node, heading_level=2)
-        title, styles = self.ast.get_heading_info(heading_node)
+    def _is_objectives_box(self, ast_node):
+        """Logic to validate a single objective box
+        (defined as a blockquoted section
+        that starts with a heading). Check that:
 
-        if not valid_head_lvl:
-            logging.error("In {0}: "
-                          "Callout box titled '{1}' must start with a "
-                          "level 2 heading".format(self.filename, title))
+        *   First child of box should a level 2 heading
+            with CSS style
+        *   Box must have only one paragraph.
+        """
+        res = True
 
-        try:
-            style = styles[0]
-        except IndexError:
+        if len(ast_node['c']) == 1:
             logging.error(
                 "In {0}: "
-                "Callout section titled '{1}' must specify "
-                "a CSS style".format(self.filename, title))
-            return False
+                "Box '{1}' should not be empty.".format(
+                    self.filename,
+                    vh.ast_to_string(ast_node['c'][0])))
+            res = False
 
-        # Track # times this style is used in any callout
-        self._callout_counts[style] += 1
-
-        # Verify style actually in callout spec
-        if style not in self.CALLOUTS:
-            spec_title = None
-            valid_style = False
-        else:
-            spec_title, _, _ = self.CALLOUTS[style]
-            valid_style = True
+        if ast_node['c'][1]['t'] != "BulletList":
+            logging.error(
+                "In {0}: "
+                "Box '{1}' should has a list as second element.".format(
+                    self.filename,
+                    vh.ast_to_string(ast_node['c'][0])))
+            res = False
 
-        has_children = self.ast.has_number_children(callout_node, minc=2)
-        if spec_title is not None and title != spec_title:
-            # Callout box must have specified heading title
+        if len(ast_node['c']) > 2:
             logging.error(
                 "In {0}: "
-                "Callout section with style '{1}' should have "
-                "title '{2}'".format(self.filename, style, spec_title))
-            valid_title = False
-        else:
-            valid_title = True
+                "Box '{1}' should has only one heading and one list.".format(
+                    self.filename,
+                    vh.ast_to_string(ast_node['c'][0])))
+            res = False
 
-        res = (valid_style and valid_title and has_children and valid_head_lvl)
         return res
 
-    def _validate_callouts(self):
-        """
-        Validate all sections that appear as callouts
+    def _is_prereq_box(self, ast_node):
```

Comment:
I removed the generic callout box validation. For every box there is a method to partialy validate it **because** they must be called from `_validate_boxes()`.